pkg/cortex/modules.go

```diff
@@ -170,6 +170,7 @@ func (t *Cortex) initDistributor(cfg *Config) (err error) {
 
 	t.server.HTTP.HandleFunc("/all_user_stats", t.distributor.AllUserStatsHandler)
 	t.server.HTTP.Handle("/api/prom/push", t.httpAuthMiddleware.Wrap(http.HandlerFunc(t.distributor.PushHandler)))
+	t.server.HTTP.Handle("/ha-tracker", t.distributor.Replicas)
```

Comment:
`Replicas` is nil is HA tracking is disabled, so this will panic.

Reply:
In general, having this be nil when HA tracking is disabled is a pain.  I would always create the struct and internalise the enabled state within that struct.